Sample_v3/Samples/SimpleChat/DetailViewController.swift

```diff
@@ -162,5 +163,12 @@ extension DetailViewController: UITableViewDataSource {
         
         return cell
     }
+    
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        if indexPath.section == tableView.numberOfSections - 1 &&
+            indexPath.row == tableView.numberOfRows(inSection: indexPath.section) - 1 {
+            controller?.loadNextMessages()
```

Comment:
@cardoso FYI: You can use the completion of this function and hook it up to some activity indicator.